### DISCUSSION TITLE
Render image on a cube side.

### DISCUSSION THREAD

**Discussion by ozh-dev** [google/filament#6386] [Q&A] [answered]
Will someone give me some advice. How to draw image on a cube side?
example: I learned how to render a random 3d model of cube from internet with android Filament, added support some gestures, and now I need render any bitmap on cube side. 

What the proper way to do it? 
I saw the samples, but they have .mat file with _fragment { }_ part, however I don't know how to write this part by my own :(
That is my second week in Filament research.

https://user-images.githubusercontent.com/8090222/207833132-c910b1e0-34af-4528-9d20-3a20702da8cd.mp4

**Comment by bejado** [comment, marked as answer]
Take a look at the [sample-textured-object](https://github.com/google/filament/tree/main/android/samples/sample-textured-object) Android sample. Inside the fragment shader, you'll want to [sample a texture](https://github.com/google/filament/blob/main/android/samples/sample-textured-object/src/main/materials/textured_pbr.mat#L39).
